spacy_experimental/biaffine_parser/arc_predicter.pyx

```diff
@@ -43,38 +44,48 @@ DEFAULT_ARC_PREDICTER_MODEL = Config().from_str(default_model_config)["model"]
     assigns=["token.head"],
     default_config={
         "model": DEFAULT_ARC_PREDICTER_MODEL,
-        "scorer": {"@scorers": "spacy-experimental.biaffine_parser_scorer.v1"}
+        "scorer": {"@scorers": "spacy-experimental.biaffine_parser_scorer.v1"},
+        "senter": None,
+        "max_length": 100,
     },
 )
 def make_arc_predicter(
     nlp: Language,
     name: str,
     model: Model,
     scorer: Optional[Callable],
+    senter: Optional[str],
+    max_length: int,
 ):
-    return ArcPredicter(nlp.vocab, model, name, scorer=scorer)
+    return ArcPredicter(nlp, model, name, max_length=max_length, scorer=scorer, senter=senter)
 
 
 class ArcPredicter(TrainablePipe):
     def __init__(
         self,
-        vocab: Vocab,
+        nlp: Language,
         model: Model,
         name: str = "arc_predicter",
         *,
+        max_length=100,
         overwrite=False,
-        scorer=parser_score
+        scorer=parser_score,
+        senter=None,
     ):
         self.name = name
         self.model = model
-        self.vocab = vocab
+        self.max_length = max_length
+        self.senter = nlp.get_pipe(senter) if senter is not None else None
+        self.vocab = nlp.vocab
         cfg = {"labels": [], "overwrite": overwrite}
         self.cfg = dict(sorted(cfg.items()))
         self.scorer = scorer
 
-    def get_loss(self, examples: Iterable[Example], scores) -> Tuple[float, Floats2d]:
+    def get_loss(self, examples: Iterable[Example], scores, lengths) -> Tuple[float, Floats2d]:
         validate_examples(examples, "ArcPredicter.get_loss")
 
+        lengths = to_numpy(lengths)
+
         def loss_func(guesses, target, mask):
```

Comment:
It's purely aesthetics to scope the loss and have it easily readable as a separate unit. Since it's only called once per batch, the overhead is really negligible.